File: android/src/main/java/com/patel/tj/RootCheckerPlugin.java
package com.patel.tj;

import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;

@CapacitorPlugin(name = "RootChecker")
public class RootCheckerPlugin extends Plugin {

    private RootChecker implementation = new RootChecker();

    @PluginMethod
    public void echo(PluginCall call) {
        String value = call.getString("value");

        JSObject ret = new JSObject();
        ret.put("value", implementation.echo(value));
        call.resolve(ret);
    }

    @PluginMethod
    public void isDeviceRooted(PluginCall call) {
        boolean isRooted = implementation.isDeviceRooted(); // Implement your rooted check logic
        JSObject ret = new JSObject();
        ret.put("isRooted", isRooted);
        call.resolve(ret);
    }
}

File: ios/Plugin/RootCheckerPlugin.swift
import Foundation
import Capacitor

/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */
@objc(RootCheckerPlugin)
public class RootCheckerPlugin: CAPPlugin {
    private let implementation = RootChecker()

    @objc func echo(_ call: CAPPluginCall) {
        let value = call.getString("value") ?? ""
        call.resolve([
            "value": implementation.echo(value)
        ])
    }

    @objc func isDeviceRooted(_ call: CAPPluginCall) {
        // Your existing Android rooted check implementation
        // ...

        // iOS jailbreak check
        let jailbreakChecker = TJJailbreakChecker()
        jailbreakChecker.isDeviceJailbroken(call)
    }


}
